# 06 - Combining datasets

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [2]:
import pandas as pd 
import random as rd 
import numpy as np



### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [3]:
serie1 = np.random.randint(1, 5, 100)
serie1
serie2 = np.random.randint(1, 4, 100)
serie3 = np.random.randint(10000, 30000, 100)

### Step 3. Let's create a DataFrame by joinning the Series by column

In [4]:
serie1 =pd.Series(serie1)
serie2 =pd.Series(serie2)
serie3 = pd.Series(serie3)
df = pd.concat([serie1, serie2, serie3], axis=1)

In [5]:
df

0  1      2
0   1  2  18235
1   1  2  27725
2   3  1  16489
3   2  1  25044
4   3  1  24876
.. .. ..    ...
95  3  1  27642
96  2  1  16928
97  1  3  29493
98  1  2  25695
99  3  1  25597

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [6]:
df.columns = ['bedrs', 'bathrs', 'price_sqr_meter']
df

bedrs  bathrs  price_sqr_meter
0       1       2            18235
1       1       2            27725
2       3       1            16489
3       2       1            25044
4       3       1            24876
..    ...     ...              ...
95      3       1            27642
96      2       1            16928
97      1       3            29493
98      1       2            25695
99      3       1            25597

[100 rows x 3 columns]

### Step 5. Create a column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [7]:
bigcolumn = pd.concat([serie1,serie2,serie3], ignore_index=True)
bigcolumn

0          1
1          1
2          3
3          2
4          3
       ...  
295    27642
296    16928
297    29493
298    25695
299    25597
Length: 300, dtype: int32

In [9]:
result = pd.DataFrame(bigcolumn)
result

0
0        1
1        1
2        3
3        2
4        3
..     ...
295  27642
296  16928
297  29493
298  25695
299  25597

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [9]:
bigcolumn.shape #nope

(300,)

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [ ]:
#no se muy bien cómo lo hice pero ya aparece con el indice así